client/src/components/JurisdictionAdmin/BatchRoundDataEntry/BatchRoundDataEntry.tsx

```diff
@@ -0,0 +1,160 @@
+import React, { useEffect, useState } from 'react'
+import styled from 'styled-components'
+import { Callout } from '@blueprintjs/core'
+
+import BatchDetails from './BatchDetails'
+import useContestsJurisdictionAdmin from '../useContestsJurisdictionAdmin'
+import { Confirm, useConfirm } from '../../Atoms/Confirm'
+import {
+  Detail,
+  List,
+  ListAndDetail,
+  ListItem,
+} from '../../Atoms/ListAndDetail'
+import {
+  IBatch,
+  IBatchResultTallySheet,
+  useBatches,
+  useRecordBatchResults,
+} from '../useBatchResults'
+import { useDebounce } from '../../../utils/debounce'
+
+const Container = styled('div')`
+  display: flex;
+  flex-direction: column;
+  gap: 12px;
+  width: 100%;
+`
+
+interface IProps {
+  electionId: string
+  jurisdictionId: string
+  roundId: string
+}
+
+const BatchRoundDataEntry: React.FC<IProps> = ({
+  electionId,
+  jurisdictionId,
+  roundId,
+}) => {
+  const batchesQuery = useBatches(electionId, jurisdictionId, roundId)
+  const contestsQuery = useContestsJurisdictionAdmin(electionId, jurisdictionId)
+  const recordBatchResults = useRecordBatchResults(
+    electionId,
+    jurisdictionId,
+    roundId
+  )
+  const { confirm, confirmProps } = useConfirm()
+
+  const [searchQuery, setSearchQuery] = useState('')
+  const [debouncedSearchQuery] = useDebounce(searchQuery)
+  const [selectedBatchId, setSelectedBatchId] = useState<IBatch['id']>()
+  const [areChangesUnsaved, setAreChangesUnsaved] = useState(false)
+
+  const batches = batchesQuery.isSuccess ? batchesQuery.data.batches : []
+  const filteredBatches = batches.filter(batch =>
+    batch.name.toLowerCase().includes(debouncedSearchQuery.toLowerCase())
+  )
+  const selectedBatch = batches.find(batch => batch.id === selectedBatchId)
+
+  // Auto-select first batch on initial load
+  useEffect(() => {
+    if (!selectedBatchId && filteredBatches.length > 0) {
+      setSelectedBatchId(filteredBatches[0].id)
+    }
+  }, [filteredBatches, selectedBatchId, setSelectedBatchId])
```

Comment:
> I'll offer suggestions to that effect (pun intended)

😆

The `useEffect` is gone! https://github.com/votingworks/arlo/pull/1661/commits/a3251e63d87bf9d027fd8c4f86ab77d4e15ae7ab

Decided to move data fetching into a wrapper component since I wasn't a huge fan of `selectedBatchId` possibly being `null` while `selectedBatch` wasn't.

In general, I do like using wrapper components for async data fetching. They can cover loading states, error states, and other preemptive return cases. And the components that they wrap and pass data to can then focus on core logic, knowing that data is defined.

Reply:
Yeah, I like the wrapper approach too! I wish there were an easy way to do that inside a component - I think Suspense could be the ticket (if we ever make the upgrade to React 18)